### DISCUSSION TITLE
When combine multiple pdf files, the inserted fields (added by javascript in Acrobat) are missing, how to solve?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#3763] [General]
When combine multiple pdf files, the inserted fields (added by javascript in Acrobat) are missing, how to solve?

[b_reduced+header.pdf](https://github.com/user-attachments/files/16554621/b_reduced%2Bheader.pdf)
[a_reduced+header.pdf](https://github.com/user-attachments/files/16554622/a_reduced%2Bheader.pdf)
[c_reduced+header.pdf](https://github.com/user-attachments/files/16554624/c_reduced%2Bheader.pdf)

Code tried:
method 1: 

def merge_pdfs(folder_path, output_path):
    # 创建一个新的 PDF 文档
    merged_document = fitz.open()

    # 遍历指定文件夹中的所有文件
    for filename in os.listdir(folder_path):
        if filename.endswith(".pdf") and "header" in filename:
            pdf_path = os.path.join(folder_path, filename)
            print(f"Merging: {pdf_path}")

            # 打开当前 PDF 文档
            current_document = fitz.open(pdf_path)

            # 将当前文档的所有页面添加到合并文档
            for page_num in range(len(current_document)):
                page = current_document[page_num]

                # 在合并文档中创建一个新页面
                new_page = merged_document.new_page(width=page.rect.width, height=page.rect.height)

                # 复制当前页面的内容到新页面
                new_page.show_pdf_page(new_page.rect, current_document, page_num)

            current_document.close()

    # 保存合并后的文档
    merged_document.save(output_path)
    merged_document.close()
    print(f"Merged PDF saved to: {output_path}")

method 2:
def merge_pdfs(folder_path, output_path):
    # 创建一个新的 PDF 文档
    merged_document = fitz.open()

    # 遍历指定文件夹中的所有文件
    for filename in os.listdir(folder_path):
        if filename.endswith(".pdf") and "header" in filename:
            pdf_path = os.path.join(folder_path, filename)
            print(f"Merging: {pdf_path}")
            # 打开当前 PDF 文档
            current_document = fitz.open(pdf_path)
            # 将当前文档的页添加到合并文档
            merged_document.insert_pdf(current_document)

    # 保存合并后的文档
    merged_document.save(output_path)
    merged_document.close()
    print(f"Merged PDF saved to: {output_path}")

**Comment by JorjMcKie** [comment]
This **_cannot_** be solved in any way!
Lookup docu about Document.bake() on how to convert annotation/fields to permanent content before merging. There will be no fields in the output technically however.
